src/pages/PageCamera.vue

```diff
@@ -45,47 +73,142 @@
 </template>
 
 <script setup>
+//imports
 import { uid } from 'quasar'
-import { reactive, onMounted, ref } from 'vue'
-import * as locales  from 'md-gum-polyfill'
+import { reactive, onMounted, ref, onBeforeUnmount } from 'vue'
+import * as locales from 'md-gum-polyfill'
+import axios, * as others from 'axios'
+import { useQuasar } from 'quasar'
 
+//data objects
 const post = reactive({
   id: uid(),
   caption: '',
   location: '',
   photo: null,
   date: Date.now()
 })
+const $q = useQuasar()
 
-const imageCaptured = reactive ({
-  captured: false
-})
+//refs
+const imageCaptured = ref(false)
+const hasCameraSupport = ref(true)
+const imageUpload = ref([])
+const locationLoading = ref(false)
 
 let video = ref()
 let canvas = ref()
 
+//methods
+const dataUrItoBlob = dataUri => {
+  let binary = atob(dataUri.split(',')[1])
+  let mimeString = dataUri.split(',')[0].split(':')[1].split(';')[0]
+  let array = []
+  for (let i = 0; i < binary.length; i++) {
+    array.push(binary.charCodeAt(i))
+  }
+  return new Blob([new Uint8Array(array)], { type: mimeString })
+}
+
 const captureImage = () => {
   canvas.value.width = video.value.getBoundingClientRect().width
   canvas.value.height = video.value.getBoundingClientRect().height
   let context = canvas.value.getContext('2d')
   context.drawImage(video.value, 0, 0, canvas.value.width, canvas.value.height)
-  imageCaptured.captured = true
+  imageCaptured.value = true
+  post.photo = dataUrItoBlob(canvas.value.toDataURL())
+  disableCamera()
+}
+
+const captureImageFallBack = file => {
+  post.photo = file
 
+  let context = canvas.value.getContext('2d')
+  let reader = new FileReader()
+  reader.onload = event => {
+    let img = new Image()
+    img.onload = () => {
+      canvas.value.width = img.width
+      canvas.value.height = img.height
+      context.drawImage(img, 0, 0)
+      imageCaptured.value = true
+    }
+    img.src = event.target.result
+  }
+  reader.readAsDataURL(file)
 }
-const initCamera = () => {
-  navigator.mediaDevices.getUserMedia({
-    video: true
-  }).then(Stream => {
-    
-    video.value.srcObject = Stream
+
+const disableCamera = () => {
+  video.value.srcObject.getVideoTracks().forEach(track => {
+    track.stop()
   })
 }
 
-onMounted( () => {
- initCamera()
+const initCamera = () => {
+  navigator.mediaDevices
+    .getUserMedia({
+      video: true
+    })
+    .then(Stream => {
+      video.value.srcObject = Stream
+    })
+    .catch(Error => {
+      hasCameraSupport.value = false
+    })
+}
+
+const getLocation = () => {
+  locationLoading.value = true
+  navigator.geolocation.getCurrentPosition(
+    position => {
+      getSityandCountry(position)
+    },
+    err => {
+      locationError()
+    },
+    { timeout: 7000 }
+  )
 }
-  
-)
+
+const getSityandCountry = async position => {
+  let apiUrl = `https://geocode.xyz/${position.coords.latitude},${position.coords.longitude}?json=1`
+  await axios
```

Comment:
lets make it this way: 

async method returns data 

and then in component something like this 

....

let location = ref(null)

location.value =  await getSityandCountry()

const locationSuccess = result => {
  post.location = location.value.data.city
  if (location.value.data.country) {
    post.location += `, ${location.value.data.country}`
  }
  locationLoading.value = false
}

// main idea is that you take your async mehod put it into api file, return value in async method and then in your UI component you get the value into your local state and use this local state in your local logic constructions